src/syscall/syscall_libc.go

```diff
@@ -80,6 +80,22 @@ func Getenv(key string) (value string, found bool) {
 	}
 }
 
+func Mmap(fd int, offset int64, length int, prot int, flags int) (data []byte, err error) {
+	addr := libc_mmap(nil, uintptr(length), int32(prot), int32(flags), int32(fd), uintptr(offset))
+	if addr == unsafe.Pointer(^uintptr(0)) {
+		return nil, getErrno()
+	}
+	return (*[30]byte)(addr)[:length], nil
```

Comment:
Well there is my bug. I meant to write:

```suggestion
	return (*[1 << 30]byte)(addr)[:length], nil
```